### DISCUSSION TITLE
Simplification of the BRouter connection

### DISCUSSION THREAD

**Discussion by WolfgangZ42** [devemux86/cruiser#238] [Q&A] [answered]
Because I wanted to demonstrate the simplicity of Cruiser to my fellow drivers, I claimed that the BRouter app only needs to be installed and never called up. 
Unfortunately, this is not true, because the very first time the BRouter app is called, the storage location is set. 

Procedure: 
1) Install and open the Cruiser App. 
2) Switching to BRouter (offline) and changing some options, Cruiser asks for the BRouter app to be installed ( image 1).
3) After installing BRouter app, options can be changed.
4) But downloading the routing data results in an error (images 2 3).
5) Only after accepting the storage location everything is ok (images 4 5).

Maybe the storage location can also be set via API? That could simplify the BRouter connection ... 

Images 1 - 5

![1 changingtoBRouter](https://github.com/devemux86/cruiser/assets/121806238/316d3dec-e861-4703-94b1-83bb3b4c85ca)  ![2 downloadBRouter](https://github.com/devemux86/cruiser/assets/121806238/4cf72c4b-4d39-43a5-9421-8ccc8aa3152c)   ![3 downloadBRouter2](https://github.com/devemux86/cruiser/assets/121806238/608dc43b-63e5-49a6-9bf5-168ecd13ed9a)  ![4 BRouter_Init](https://github.com/devemux86/cruiser/assets/121806238/f386b226-2f14-45d1-ba63-f7ca482eb039)  ![5 BRouter_Init](https://github.com/devemux86/cruiser/assets/121806238/4fd9677e-ba1d-4ae0-af8b-db6345f9b1bc)

**Comment by devemux86** [comment]
@WolfgangZ42 thanks, very good observation!

I need to check how we can improve this situation. :slightly_smiling_face: 

cc @afischerdev

**Reply by WolfgangZ42** [reply]
Thanks for looking after this - I just would like that my statement
![image](https://github.com/devemux86/cruiser/assets/121806238/92e9a192-306d-44d3-b861-654c8a1db136)
becomes true.

**Comment by devemux86** [comment]
I want to publicly thank @afischerdev for the great support! :+1: 
He is always very helpful when we ask about BRouter operation.

---

With next BRouter app update, when we open BRouter download manager from within Cruiser, it will work immediately.
(no need to open the BRouter app at all first)

Bonus: with the next BRouter app update, the BRouter download manager can also be used in portrait mode!

**Reply by Ciutnik** [reply]
Maybe with the consent of the BRouter developer, it would be possible to update the .rd5 segments from the Cruiser without invoking the BRouter app? In the same window as maps or GraphHopper files? It's just a matter of consistency in the Cruiser's styling, which may be of some importance to novices.

Because I understand that to select the areas of interest to us, we need to call up the BRouter window - at least for now.

**Reply by MotoUKRider** [reply]
Random Sunday afternoon musing... I've always considered the colour-scheme for the tiles in BRouter a little "odd".  A grey square doesn't scream "updated area" to me - it looks kinda boring, ie: unchanged since the last time it was downloaded.  Blue or Yellow seem more eye-catching.

**Reply by devemux86** [reply]
@Ciutnik
BRouter files are in its own media folder. In latest Androids, apps can only write in their folders (except file managers).
The best alternative right now seems to be calling the BRouter downloader from inside Cruiser.

@MotoUKRider 
They have a topic with ideas for improving the user experience.
Can you also write there, so such suggestions are not lost here?
- https://github.com/abrensch/brouter/issues/346

**Reply by Ciutnik** [reply]
> In latest Androids, apps can only... ... ...

As they say, better is the enemy of good... I'm already starting to get lost in these ever-increasing limitations "for the benefit of users". We'll have iAndroid soon 🙄

**Comment by devemux86** [comment, marked as answer]
> With next BRouter app update, when we open BRouter download manager from within Cruiser, it will work immediately.
(no need to open the BRouter app at all first)

Available in [Cruiser 3.0.25](https://play.google.com/store/apps/details?id=gr.talent.cruiser) with [BRouter 1.7.1](https://play.google.com/store/apps/details?id=btools.routingapp).
